core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/LikeExpression.kt

```diff
@@ -2,31 +2,37 @@ package ch.ergon.dope.resolvable.expression.unaliased.type.relational
 
 import ch.ergon.dope.DopeQueryManager
 import ch.ergon.dope.resolvable.expression.TypeExpression
-import ch.ergon.dope.resolvable.expression.unaliased.type.Field
 import ch.ergon.dope.resolvable.expression.unaliased.type.toDopeType
 import ch.ergon.dope.resolvable.operator.InfixOperator
 import ch.ergon.dope.validtype.BooleanType
 import ch.ergon.dope.validtype.StringType
-import ch.ergon.dope.validtype.ValidType
 
 class LikeExpression(
-    left: Field<out ValidType>,
+    left: TypeExpression<StringType>,
     right: TypeExpression<StringType>,
 ) : TypeExpression<BooleanType>, InfixOperator(left, "LIKE", right) {
     override fun toDopeQuery(manager: DopeQueryManager) = toInfixDopeQuery(manager = manager)
 }
 
-fun Field<out ValidType>.isLike(right: TypeExpression<StringType>) = LikeExpression(this, right)
+fun TypeExpression<StringType>.isLike(right: TypeExpression<StringType>) = LikeExpression(this, right)
 
-fun Field<out ValidType>.isLike(right: String) = isLike(right.toDopeType())
+fun TypeExpression<StringType>.isLike(right: String) = isLike(right.toDopeType())
+
+fun String.isLike(right: TypeExpression<StringType>) = toDopeType().isLike(right)
```

Comment:
please add some buildTest, where you pass more complex type-expression (e.g. concat, like it is described in the ticket)